pkg/services/ngalert/state/state.go

```diff
@@ -138,8 +138,10 @@ func (a *State) Resolve(reason string, endsAt time.Time) {
 }
 
 // Maintain updates the end time using the most recent evaluation.
-func (a *State) Maintain(interval int64, evaluatedAt time.Time) {
+func (a *State) Maintain(interval int64, evaluatedAt time.Time) time.Time {
```

Comment:
NIT: I don't think we need to change the signature of `Maintain` to return `EndsAt` considering we have access to that via `state.EndsAt` at the point in the code that we're doing the logging. I don't mind either way though, if you have a stronger opinion here, that's fine with me and keep as is.

Reply:
Ah, so the reason I did this is so we wouldn't need to do `oldEndsAt := state.EndsAt` each time. Do you think that's better?